### ISSUE TITLE
Limit commits used for snapshot releases to those that passed all checks

### ISSUE THREAD

**Issue by ScottTodd** [google/iree#6314]
The `Schedule snapshot release` workflow ([source](https://github.com/google/iree/blob/main/.github/workflows/schedule_snapshot_release.yml)) is run on a schedule or when manually triggered: https://github.com/google/iree/blob/871a15995432fd30444dd9d39ae1825a20d2ccec/.github/workflows/schedule_snapshot_release.yml#L1-L7

However, the workflow currently picks the latest commit, rather than the latest commit _that passed all checks_. Our presubmit coverage and overall test hygiene are both pretty good, but being a bit smarter about commit selection will help us avoid release builds that are doomed to fail or automatically pushing snapshot releases that have detectable issues.

We should pick the latest commit where all checks we expect to run should be finished running with successful results. Additionally, if a release was already built for the chosen commit (check git tags?) then the workflow should abort and try again when next scheduled.

We could also add parameters to the workflow to allow for manually triggering at a specific commit and/or overriding any skip behavior.

We may be able to use native functionality, reuse an existing action (like https://github.com/nrwl/last-successful-commit-action), or reference what other projects do for their own triggering.

**Comment by GMNGeoffrey** [comment]
@not-jenni are you working on this? (no worries if not, I just had the impression you were :slightly_smiling_face: )

**Comment by not-jenni** [comment]
Yes, thank you!

**Comment by GMNGeoffrey** [comment]
@not-jenni are you still working on this? It looks like it may have gotten lost and it's pretty important

**Comment by not-jenni** [comment]
Thanks for bumping this - I started working on this and think the change will look somewhat like this commit https://github.com/google/iree/commit/228761e29392097bb5e8ac0d92cd0baba91e96c8, but I got stuck trying to figure out how to test it on my fork because it doesn’t have presubmits that can fail.  I think one option for testing is to somehow create an action on my fork that can make a commit look like a failure to the action grabbing the commit, I will look more into that now

**Comment by GMNGeoffrey** [comment]
Yeah you can have actions on your fork. You just need to enable them (https://github.com/not-jenni/iree/actions). Actions are enabled for my fork. Then you can just create an action that runs `exit 1`. I'm not sure whether it needs to run on the "push" event in order to post statuses on the commit

**Comment by not-jenni** [comment]
This worked for testing, thank you for the help!
